[tutorial] Ch 3-6 questions redone (#1852)
withastro/docs: src/pages/en/tutorial/3-components/4.md

### Test your knowledge

1. When does Astro run any JavaScript written in a component's frontmatter?

>   <MultipleChoice>
>     <Option>
>      Astro never runs JavaScript
>     </Option>
>     <Option isCorrect>
>       at build-time
>     </Option>
>     <Option>
>        When a visitor clicks a button
>     </Option>
>   </MultipleChoice>

Reviewer: Indentation!

```suggestion
    <MultipleChoice>
      <Option>
       Astro never runs JavaScript
      </Option>
      <Option isCorrect>
        at build-time
      </Option>
      <Option>
         When a visitor clicks a button
      </Option>
    </MultipleChoice>
```

> 2. Optionally, Astro can send JavaScript to the browser for:
>   <MultipleChoice>
>     <Option>
>       `<script>` tags
>     </Option>
>     <Option>
>       build time
>     </Option>
>     <Option isCorrect>
>        client-side interactivity
>     </Option>
>   </MultipleChoice>
> 

Author: ```suggestion
2. Optionally, Astro can send JavaScript to the browser to allow:
  <MultipleChoice>
    <Option>
      users to click page links
    </Option>
    <Option>
      faster load times
    </Option>
    <Option isCorrect>
       client-side interactivity
    </Option>
  </MultipleChoice>

```

>   <MultipleChoice>
>     <Option>
>       users to click page links
>     </Option>
>     <Option>
>       faster load times
>     </Option>
>     <Option isCorrect>
>        client-side interactivity
>     </Option>
>   </MultipleChoice>

Reviewer: ```suggestion
    <MultipleChoice>
      <Option>
        users to click page links
      </Option>
      <Option>
        faster load times
      </Option>
      <Option isCorrect>
        client-side interactivity
      </Option>
    </MultipleChoice>
```

> 3. Only the JavaScript written or imported here will not be discarded after your site is built:

Reviewer: “Only… not…” is a confusing formulation. Can we rewrite?

```suggestion
3. The JavaScript written or imported here will be sent to a user’s browser:
```

(Not entirely happy with this, can we do better? 😅)

> 3. Only the JavaScript written or imported here will not be discarded after your site is built:

Reviewer: “Only… not…” is a confusing formulation. Can we rewrite?

```suggestion
3. The JavaScript written or imported here will be sent to a user’s browser:
```

(Not entirely happy with this, can we do better? 😅)

> 1. When does Astro run any JavaScript written in a component's frontmatter?
>   <MultipleChoice>
>     <Option>
>       `<script>` tags
>     </Option>
>     <Option isCorrect>
>       build time
>     </Option>
>     <Option>
>        client-side interactivity
>     </Option>
>   </MultipleChoice>
> 
> 2. Optionally, Astro can send JavaScript to the browser to allow:
>   <MultipleChoice>
>     <Option>
>       users to click page links
>     </Option>
>     <Option>
>       faster load times
>     </Option>
>     <Option isCorrect>
>        client-side interactivity
>     </Option>
>   </MultipleChoice>
> 
> 3. Only the JavaScript written or imported here will not be discarded after your site is built:
>   <MultipleChoice>
>     <Option isCorrect>
>       `<script>` tags
>     </Option>
>     <Option>
>       build time
>     </Option>
>     <Option>
>        client-side interactivity
>     </Option>
>   </MultipleChoice>

Author: ```suggestion
1. When does Astro run any JavaScript written in a component's frontmatter?
  <MultipleChoice>
    <Option>
     Astro never runs JavaScript
    </Option>
    <Option isCorrect>
      at build-time
    </Option>
    <Option>
       When a visitor clicks a button
    </Option>
  </MultipleChoice>

2. Optionally, Astro can send JavaScript to the browser to allow:
  <MultipleChoice>
    <Option>
      users to click page links
    </Option>
    <Option>
      faster load times
    </Option>
    <Option isCorrect>
       client-side interactivity
    </Option>
  </MultipleChoice>

3. Only the JavaScript written or imported here will not be discarded after your site is built:
  <MultipleChoice>
    <Option isCorrect>
      in `<script>` tags
    </Option>
    <Option>
      between a `.astro` file's code fences
    </Option>
    <Option>
      in `src/components/`
    </Option>
  </MultipleChoice>
```

>   <MultipleChoice>
>     <Option isCorrect>
>       in `<script>` tags
>     </Option>
>     <Option>
>       between a `.astro` file's code fences
>     </Option>
>     <Option>
>       in `src/components/`
>     </Option>
>   </MultipleChoice>

Reviewer: ```suggestion
    <MultipleChoice>
      <Option isCorrect>
        in `<script>` tags
      </Option>
      <Option>
        between a `.astro` file's code fences
      </Option>
      <Option>
        in `src/components/`
      </Option>
    </MultipleChoice>
```
